addons/website/tests/test_lang_url.py

```diff
@@ -28,30 +30,40 @@ def test_02_url_redirect(self):
         url = '/fr_WHATEVER/contactus'
         r = self.url_open(url)
         self.assertEqual(r.status_code, 200)
-        self.assertTrue(r.url.endswith('/fr/contactus'), "fr_WHATEVER should be forwarded to 'fr_FR' lang as closest match")
+        self.assertTrue(r.url.endswith('/fr/contactus'), f"fr_WHATEVER should be forwarded to 'fr_FR' lang as closest match, url: {r.url}")
 
         url = '/fr_FR/contactus'
         r = self.url_open(url)
         self.assertEqual(r.status_code, 200)
-        self.assertTrue(r.url.endswith('/fr/contactus'), "lang in url should use url_code ('fr' in this case)")
+        self.assertTrue(r.url.endswith('/fr/contactus'), f"lang in url should use url_code ('fr' in this case), url: {r.url}")
 
     def test_03_url_cook_lang_not_available(self):
         """ An activated res.lang should not be displayed in the frontend if not a website lang. """
         self.website.language_ids = self.env.ref('base.lang_en')
         r = self.url_open('/fr/contactus')
-        self.assertTrue('lang="en-US"' in r.text, "french should not be displayed as not a frontend lang")
+        doc = lxml.html.document_fromstring(r.text)
```

Comment:
It is kinda useless diff, I changed this part because the error reporting was `False is not True` which was a pain to work with. To always parse the body is clearly not as fast as `str.__contains__`. I'll change the code as follow:

```python
    if 'lang="en-US"' not in r.text:
        doc = lxml.html.document_fromstring(r.text)
        ...
```

so it is fast when there is no error and there is a proper error reporting when it fails.